test/Microsoft.Framework.Caching.Memory.Tests/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -209,5 +209,29 @@ public void RemoveAndReAddFromCallbackWorks()
             result = cache.Get(key);
             Assert.Same(obj2, result);
         }
+
+        [Fact]
+        public void SetGetAndRemoveWorks_WithNonStringKeys()
+        {
+            var cache = CreateCache();
+            var obj = new object();
+            var key = new Person { Id = 10, Name = "Mike" };
+
+            var result = cache.Set(key, obj);
+            Assert.Same(obj, result);
+
+            result = cache.Get(key);
+            Assert.Same(obj, result);
```

Comment:
Unless I am missing something, this is succeeding merely because you used the same instance of Person to set and retrieve the entry. I don't think it is very representative of what needs to happen in the real world. E.g. Person would need to implement structural equality for `Equals()` and compute `GetHashCode()` based on its properties, so that you can use a different instance of it with the same property values to retrieve the same entry. 

I would recommend you do that explicitly and use different instance for `Set()`, `Get()` and `Remove()` or that you just swap to three anonymous type instances of the same structure and values (and the compiler will implement structural equality for you).